cmd/kubernetes-discovery/discoverysummarizer/types.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package discoverysummarizer
+
+// List of servers from which group versions should be summarized.
+type FederatedServerList struct {
```

Comment:
How about `cmd/kubernetes-discovery/discoverysummarizer/apis/config/v1alpha1/types.go`?
By componentconfig, my expectation is that it specifies the command line arguments for various binaries which is not what this does.
The structs here specify the structure of the config file that is passed while running the server.